.github/workflows/tests.yaml

```diff
@@ -307,6 +313,20 @@ jobs:
           # Propagate the Bazel exit code.
           exit $bazel_exit
 
+      # Run in the clang-tidy config. This is done as part of tests so that we
+      # aren't duplicating bazel/llvm setup.
+      #
+      # The `-k` flag is used to print all clang-tidy errors.
+      - name: clang-tidy
+        if:
+          steps.filter.outputs.has_code == 'true' && matrix.os == 'ubuntu-22.04'
+          && matrix.build_mode == 'clang-tidy'
+        env:
+          TARGETS_FILE: ${{ runner.temp }}/targets
+        run: |
+          bazelisk build --config=clang-tidy -k \
+            --target_pattern_file=$TARGETS_FILE
```

Comment:
Hmm, I'm a bit worried we'll want similar retry logic here as we have for tests. But maybe not?

I'm happy to try it out and see how it goes.

Reply:
I *think* we don't need retry here due to the difference in what's executing, although I may be misunderstanding the source of failures.